### PR TITLE
Gokit/payment

### PR DESCRIPTION
**Pull request by philwinder**

### CODE REVIEW

File: payment/endpoints.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	return Endpoints{
+		AuthoriseEndpoint:  MakeAuthoriseEndpoint(s),
+	}
+}
+
+// MakeListEndpoint returns an endpoint via the given service.
+func MakeAuthoriseEndpoint(s Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(authoriseRequest)
+		fmt.Println("STUB:", req)
```
Comment (peterbourgon) [inline payment/endpoints.go]: Whoopsy

File: payment/service.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+type service struct {
+}
+
+func (s*service) Authorise() (Authorisation) {
+	return Authorisation{
+		Authorised: true,
+	}
+}
```
Comment (peterbourgon) [inline payment/service.go:25]: This file needs a gofmtting

File: payment/transport.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	return r
+}
+
+func encodeError(_ context.Context, err error, w http.ResponseWriter) {
+	code := http.StatusInternalServerError
+	//switch err {
+	//case ErrNotFound:
+	//	code = http.StatusNotFound
+	//}
```
Comment (peterbourgon) [inline payment/transport.go]: Delete me

File: payment/transport.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+// MakeHTTPHandler mounts the endpoints into a REST-y HTTP handler.
+func MakeHTTPHandler(ctx context.Context, e Endpoints, logger log.Logger) http.Handler {
+	r := mux.NewRouter().StrictSlash(false)
+	options := []httptransport.ServerOption{
+		httptransport.ServerErrorLogger(logger),
+		httptransport.ServerErrorEncoder(encodeError),
+	}
+
+	r.Methods("GET").Path("/paymentAuth").Handler(httptransport.NewServer(
```
Comment (peterbourgon) [inline payment/transport.go]: Shouldn't this be POST? (Maybe this is outside the scope of this PR.)

File: payment/transport.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+		ctx,
+		e.AuthoriseEndpoint,
+		decodeAuthoriseGetRequest,
+		encodeAuthoriseGetResponse,
```
Comment (peterbourgon) [inline payment/transport.go]: Why do you call it xxcodeAuthorise**Get**Rexxx, instead of just xxcodeAuthoriseRexxx?

File: payment/transport.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+		"error":       err.Error(),
+		"status_code": code,
+		"status_text": http.StatusText(code),
+	})
+}
+
+func decodeAuthoriseGetRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	return authoriseRequest{
+	}, nil
+}
```
Comment (peterbourgon) [inline payment/transport.go:44]:
``` go
func decodeAuthoriseRequest(context.Context, *http.Request) (interface{}, error) {
    return authoriseRequest{}, nil
}
```

File: payment/transport.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+// encodeGetResponse is distinct from the generic encodeResponse because we need
+// to special-case when the getResponse object contains a non-nil error.
+func encodeAuthoriseGetResponse(ctx context.Context, w http.ResponseWriter, response interface{}) error {
```
Comment (peterbourgon) [inline payment/transport.go]: Comment does not match function.

File: payment/endpoints.go [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+		return authoriseResponse{Authorisation: authorisation}, nil
+	}
+}
+
+type authoriseRequest struct {
+}
+
+type authoriseResponse struct {
+	Authorisation Authorisation
+}
```
Comment (peterbourgon) [inline payment/endpoints.go:35]: This file needs a gofmtting

**Comment by peterbourgon** [comment]
A few nits, a few minor-ish issues, but with those fixed, LGTM.

File: payment/endpoints.go (hunk 1 above)
Comment (philwinder) [inline payment/endpoints.go]: I wanted to keep this so when I add something useful to the request, the code is there to handle it. By removing the req I would have to remove the interface, the parameter, etc. Which I didn't want to do because I wanted it to stay as similar to yours as possible.

File: payment/transport.go (hunk 5 above)
Comment (philwinder) [inline payment/transport.go]: In case I had other methods in there. But I guess in that case, it should be named appropriately. You're right.

File: payment/endpoints.go (hunk 1 above)
Comment (peterbourgon) [inline payment/endpoints.go]: Well, you need to remove the Println, at least. If you just want to get around the declared-but-not-used error, you can do

``` go
_ = request.(authoriseRequest)
```

**Comment by philwinder** [comment]
Intellij has a `gofmt on commit` option. :+1:
